e2e/specs/results-viewer.spec.ts

```diff
@@ -0,0 +1,189 @@
+    await page.locator('#manualEntrySporePresenceInUrineTestid_3').check();
+    await page.locator('#manualEntryTrichomonasVaginalisPresenceInUrineSedimentByLightMicroscopyid_0').check();
+    await page.locator('#manualEntryCrystalsTypeInUrineSedimentByLightMicroscopyTestid_3').check();
+    await page.locator('#manualEntryStoolCultureBacterialid').fill('no signs');
+    await page.locator('#manualEntryStoolExamid').selectOption('Cestode Infection');
+    await page.locator('#manualEntryStoolMicroscopyWithConcentrationid').fill('no signs');
+    await page.locator('#manualEntryKinyounsStainForCoccidiansid').fill('no signs');
+    await page.locator('#manualEntryStoolFatTestSemiQuantitativeid_3').check();
+    await page.locator('#manualEntryStoolTestforReducingSubstanceid_0').check();
+    await page.locator('#manualEntryFecalOccultBloodTestid_2').check();
+  });
+
+  await test.step('And I save the form', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success toast notification', async () => {
+    await expect(page.getByText('The form has been submitted successfully.')).toBeVisible();
+  });
+
+  await test.step('When I reload the viewer page', async () => {
+    await resultsViewerPage.goTo(patient.uuid);
+  });
+
+  await test.step('Then I should see 12 and 15 as white and red blood cells test results displayed respectively', async () => {
+    await expect(page.getByRole('link', { name: /white blood cells/i })).toBeVisible();
+    await expect(page.getByText('12', { exact: true })).toBeVisible();
+    await expect(page.getByRole('link', { name: /red blood cells/i })).toBeVisible();
+    await expect(page.getByText('15', { exact: true })).toBeVisible();
+  });
+
+  await test.step('When I navigate to `Visits` page', async () => {
+    visitsPage.goTo(patient.uuid);
+  });
+
+  await test.step('And I go to the `All encounters` tab', async () => {
+    await page.getByRole('tab', { name: /all encounters/i }).click();
+  });
+
+  await test.step('Then I should see the newly added lab test result in the list', async () => {
+    await expect(
+      page.getByRole('cell', { name: /laboratory test results/i }).getByText('Laboratory Test Results'),
+    ).toBeVisible();
+  });
+
+  await test.step('When I launch the overflow menu of the created lab test result', async () => {
+    await page
+      .getByRole('button', { name: /options/i })
+      .nth(0)
+      .click();
+  });
+
+  await test.step('And I click on the `Edit` button', async () => {
+    await page.getByRole('menuitem', { name: /edit this encounter/i }).click();
+  });
+
+  await test.step('And I change the white and red blood cells test results to `13` and `16` respectively', async () => {
+    await page.locator('#ManualInputWhiteBloodCellsid').fill('13');
+    await page.locator('#ManualEntryRedBloodCellsid').fill('16');
```

Comment:
@kdaud finally, I think this would be the reason why I need to only fill a few inputs as opposed to trying to fill everything. 
I am skeptical about having very similar, lengthy. and repetitive codes. 
I'll reduce the entries and then clear the inputs before filling.

Cheers